error/s2n_errno.h

```diff
@@ -257,6 +257,7 @@ typedef enum {
     S2N_ERR_ASYNC_APPLY_WHILE_INVOKING,
     S2N_ERR_ASYNC_ALREADY_APPLIED,
     S2N_ERR_T_USAGE_END,
+    S2N_ERR_INVALID_HELLO_RETRY,
```

Comment:
This error is in the wrong place. You put it after S2N_ERR_T_USAGE_END.

Also, do you think it's a usage error or an internal error? Or is there some RFC basis for it that would make it a protocol error?

Reply:
Offline discussion: Lets be more specific and call it `S2N_ERR_INVALID_HELLO_RETRY` and we will place it in the S2N_ERR_T_PROTO category.